tmt/steps/provision/__init__.py

```diff
@@ -1646,6 +1663,60 @@ def essential_requires(cls) -> list['tmt.base.Dependency']:
 
         return []
 
+    def acquire_log(self, logname: str) -> Optional[str]:
+        """
+        Fetch and return content of a log.
+
+        :param logname: name of the log.
+        :returns: content of the log, or ``None`` if the log cannot be retrieved.
+        """
+        raise NotImplementedError
+
+    def store_log(self, path: Path, content: str, logname: Optional[str] = None) -> None:
```

Comment:
This method can move to `GuestLog` class, right? There's `log.fetch()` followed by `store_log()`, why not have e.g. `save()` method accepting this arguments, which would then call `self.fetch()` and `self.store()`?